nbclient/client.py

```diff
@@ -832,6 +856,24 @@ def handle_comm_msg(self, outs, msg, cell_index):
             self.widget_state.setdefault(content['comm_id'], {}).update(data['state'])
             if 'buffer_paths' in data and data['buffer_paths']:
                 self.widget_buffers[content['comm_id']] = self._get_buffer_data(msg)
+        # There are cases where we need to mimic a frontend, to get similar behaviour as
+        # when using the Output widget from Jupyter lab/notebook
+        if msg['msg_type'] == 'comm_open' and msg['content'].get('target_name') == 'jupyter.widget':
```

Comment:
Maybe we need to register comm_open and comm_msg handlers per comm target, eventually.

So instead of doing `msg['content'].get('target_name') == 'jupyter.widget'` we would do `msg['content'].get('target_name') in comm_open_handlers`.